### PR TITLE
RAIN-2287 optimise the localisation service call in PDF service

### PR DESCRIPTION
**Pull request by rohit-eGov**

### CODE REVIEW

File: pdf-service/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,13 @@
 # Changelog
 All notable changes to this module will be documented in this file.
 
-## 1.1.0 - 2020-06-19
+## 1.1.2 - 2021-02-26
+- Updated domain name in application.properties
```
Comment (tarunlalwani) [inline pdf-service/CHANGELOG.md:5]: Changelog is not updated

File: pdf-service/package.json [hunk 2]
```diff
@@ -64,7 +64,8 @@
     "eslint": "^3.1.1",
     "mustache": "^3.0.1",
     "nodemon": "^1.19.4",
-    "pdfmake": "^0.1.62"
+    "pdfmake": "^0.1.62",
+    "node-fetch": "^2.6.1"
```
Comment (tarunlalwani) [inline pdf-service/package.json:68]: Don't add a new dependency, remove this and use `axios` instead

File: pdf-service/src/index.js [hunk 3]
```diff
@@ -500,6 +502,64 @@ app.post(
...
+        }
+      });
+      res.status(200);
+      res.json({
+          ResponseInfo: requestInfo,
+          unregisteredLocalisationCodes: unregisteredLocalisationCodes,
+        });
+    } catch (error) {
+      logger.error(error.stack || error);
+      res.status(500);
```
Comment (tarunlalwani) [inline pdf-service/src/index.js:553]: We should never throw any `500` errors, all errors should also be `400` and use the format that we use in other services. Check the whole service for fixes

File: pdf-service/src/utils/commons.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { httpRequest } from "../api/api";
+const fetch = require("node-fetch");
```
Comment (tarunlalwani) [inline pdf-service/src/utils/commons.js:1]: Use `axios`

File: pdf-service/src/utils/commons.js [hunk 5]
```diff
@@ -53,6 +44,47 @@ export const findAndUpdateLocalisation = async (
     defaultTenant
   ).split(".")[0];
 
+
+  let url = egovLocHost + '/localization/messages/v2/_search';
```
Comment (tarunlalwani) [inline pdf-service/src/utils/commons.js:48]: Take the path also from a config

File: pdf-service/src/utils/commons.js [hunk 6]
```diff
@@ -53,6 +44,47 @@ export const findAndUpdateLocalisation = async (
...
+
+  var options = {
+    method: 'POST',
+    body: JSON.stringify(request),
+    headers: {
+        'Content-Type': 'application/json'
+    }
+  }
+
+  let response = await fetch(url, options);
```
Comment (tarunlalwani) [inline pdf-service/src/utils/commons.js:70]: use `axios`

**Review by tarunlalwani — CHANGES_REQUESTED** [review]

File: pdf-service/src/utils/commons.js (hunk 6 above)
Comment (rohit-eGov) [reply pdf-service/src/utils/commons.js]: removed unnecessary dependency and use axios for https request call

File: pdf-service/src/utils/commons.js (hunk 5 above)
Comment (rohit-eGov) [reply pdf-service/src/utils/commons.js]: done

File: pdf-service/src/utils/commons.js (hunk 4 above)
Comment (rohit-eGov) [reply pdf-service/src/utils/commons.js]: removed unnecessary dependency and use axios for https request call

File: pdf-service/src/index.js (hunk 3 above)
Comment (rohit-eGov) [reply pdf-service/src/index.js]: updated error status

Comment (rohit-eGov) [reply pdf-service/src/index.js]: updated error status

File: pdf-service/package.json (hunk 2 above)
Comment (rohit-eGov) [reply pdf-service/package.json]: removed unnecessary dependency and use axios for https request call

File: pdf-service/CHANGELOG.md (hunk 1 above)
Comment (rohit-eGov) [reply pdf-service/CHANGELOG.md:8]: updated

**Review by tarunlalwani — APPROVED** [review]

**Pull request merged**